### PR TITLE
update IoT unit test names to unify

### PR DESCRIPTION
**Pull request by danewalton**
Some names of unit tests were inconsistent. Patching those up.

### CODE REVIEW

File: sdk/tests/iot/hub/main.c [hunk 1]
```diff
@@ -15,12 +15,12 @@ int main()
 {
   int result = 0;
 
-  result += test_iot_hub_c2d();
   result += test_iot_hub_client();
-  result += test_iot_sas_token();
-  result += test_iot_hub_telemetry();
-  result += test_az_iot_hub_client_twin();
-  result += test_iot_hub_methods();
+  result += test_iot_hub_client_c2d();
```
Comment (momuno) [inline sdk/tests/iot/hub/main.c:19]: Why not use test_az_iot_hub_client_x for these?
I notices that naming structure is used in these files, but it is for the static functions.  This non-static is the exception.

Comment (danewalton) [reply sdk/tests/iot/hub/main.c:19]: Yea my original logic was api's shouldn't be prefix'd with `az_` unless they were public but these are prefix'd with `test_`. I blame my lack of coffee this morning. Will change

Comment (momuno) [reply sdk/tests/iot/hub/main.c]: are you thinking the static functions lose the az, and the public one gains it?  Or something else?

**Pull request merged**